test/passes/optimize-instructions_all-features.txt

```diff
@@ -3417,14 +3417,73 @@
    )
   )
  )
- (func $optimize-boolean (param $x i32)
+ (func $optimize-boolean (param $x i32) (param $y i64)
   (drop
    (select
     (i32.const 1)
     (i32.const 2)
     (local.get $x)
    )
   )
+  (drop
+   (i32.and
+    (local.get $x)
+    (i32.const 1)
+   )
+  )
+  (drop
+   (i32.eqz
+    (i32.and
+     (local.get $x)
+     (i32.const 1)
+    )
+   )
+  )
+  (drop
+   (i64.eq
+    (i64.and
+     (local.get $y)
+     (i64.const 1)
+    )
+    (i64.const 1)
+   )
+  )
+  (drop
+   (i64.ne
+    (i64.and
+     (local.get $y)
+     (i64.const 1)
+    )
+    (i64.const 1)
+   )
+  )
```

Comment:
It looks like these two optimizations didn't trigger for some reason.

Reply:
Yes. It was expected. On description I mentioned this PR depend on this another: #2879

Reply:
But #2879 pretty complex and should be double check. Not sure it ready yet